security/authorization/util/util_test.go

```diff
@@ -0,0 +1,131 @@
+		decls.NewIdent("connection.uri_san_peer_certificate", decls.String, nil)}
+	env, _ := cel.NewEnv() // External Evaluation env does not have to be built the same
+	for _, test := range []struct {
+		desc                      string
+		expectedEvaluationOutcome bool
+		expectedParsingError      bool
+		expectedEvaluationError   bool
+		expr                      string
+		authzArgs                 map[string]interface{}
+	}{
+		{
+			desc:                      "1: use a policy with startsWith primitive and returns a match.",
+			expectedEvaluationOutcome: true,
+			expr:                      "request.url_path.startsWith('/pkg.service/test')",
+			authzArgs:                 map[string]interface{}{"request.url_path": "/pkg.service/test"},
+		},
+		{
+			desc:                      "2: use a policy with == primitive and returns a match.",
+			expectedEvaluationOutcome: true,
+			expr:                      "connection.uri_san_peer_certificate == 'cluster/ns/default/sa/admin'",
+			authzArgs:                 map[string]interface{}{"connection.uri_san_peer_certificate": "cluster/ns/default/sa/admin"},
+		},
+		{
+			desc:                      "3: use a policy with startsWith primitive and returns no match.",
+			expectedEvaluationOutcome: false,
+			expr:                      "request.url_path.startsWith('/pkg.service/test')",
+			authzArgs:                 map[string]interface{}{"request.url_path": "/source/pkg.service/test"},
+		},
+		{
+			desc:                      "4: use a policy with startsWith and == primitive and returns a match.",
+			expectedEvaluationOutcome: true,
+			expr:                      "request.url_path == '/pkg.service/test' && connection.uri_san_peer_certificate == 'cluster/ns/default/sa/admin'",
+			authzArgs: map[string]interface{}{"request.url_path": "/pkg.service/test",
+				"connection.uri_san_peer_certificate": "cluster/ns/default/sa/admin"},
+		},
+		{
+			desc:                 "5: use a policy that has a field not present in the environment.",
+			expectedParsingError: true,
+			expr:                 "request.source_path.startsWith('/pkg.service/test')",
+			authzArgs:            map[string]interface{}{"request.url_path": "/pkg.service/test"},
+		},
+		{
+			desc:                    "6: use an argument that has a filed not included in the environment.",
+			expectedEvaluationError: true,
+			expr:                    "request.url_path.startsWith('/pkg.service/test')",
+			authzArgs:               map[string]interface{}{"request.source_path": "/pkg.service/test"},
+		},
+	} {
+		test := test
+		t.Run(test.desc, func(t *testing.T) {
+			expected, expr, authzArgs := test.expectedEvaluationOutcome, test.expr, test.authzArgs
+			checked, err := convertStringToCheckedExpr(expr, declarations)
+			if test.expectedParsingError {
+				if err == nil {
+					t.Fatalf("lack of error in conversion %v", err)
+				}
+				return
+			}
+			if err != nil {
+				t.Fatalf("error in conversion %v", err)
+			}
```

Comment:
For the above logic, can we do:
```
if want, got := test.expectedParsingError, err != nil; got != want {
		return t.Fatalf("Error in conversion, test.expectedParsingError=%v, got %v", want, got)
	}
```
Same applied to the gotErr below.
Also if we are using t.Fatalf, the first letter shall be capital letter because program will pause.

Reply:
This does not exit correctly if we are expecting an error and we get an error. In this case we need not throw and error but return early as to not allow next steps in process

Reply:
Ah yeah, good catch. And thanks for the change!